libs/components/src/lib/selectable-box/selectable-box.ts

```diff
@@ -0,0 +1,91 @@
+import { FoundationElement } from '@microsoft/fast-foundation';
+import { attr } from '@microsoft/fast-element';
+import type { Connotation, Size } from '../enums.js';
+
+/**
+ * Types of selectable-box connotation.
+ *
+ * @public
+ */
+export type SelectableBoxConnotation = Connotation.Accent | Connotation.CTA;
+
+/**
+ * Types of selectable-box connotation.
+ *
+ * @public
+ */
+export type SelectableBoxSize = Size.Normal | Size.Condensed;
+
+export type SelectableBoxControl = 'checkbox' | 'radio';
+
+
+/**
+ * Base class for selectable-box
+ *
+ * @public
+ */
+export class SelectableBox extends FoundationElement {
+	@attr({attribute: 'aria-label'}) override ariaLabel: string | null = null;
+	@attr({attribute: 'aria-labelledby'}) ariaLabelledby: string | null = null;
+	/**
+	 * The connotation the selectable box should have.
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: connotation
+	 */
+	@attr connotation?: SelectableBoxConnotation;
+
+	/**
+	 * The type of control the box should have: checkbox or radio.
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: connotation
+	 */
+	@attr control?: SelectableBoxControl;
+
+	/**
+	 * The amount of spacing the selectable box should have.
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: spacing
+	 */
+	@attr size?: SelectableBoxSize;
+
+	/**
+	* Removes the padding around the box's slot content
+	*
+	* @public
+	* HTML Attribute: no-padding
+	*/
+	@attr({ mode: 'boolean' }) tight = false;
+
+	/**
+	* Controls the selected state of the box
+	*
+	* @public
+	* HTML Attribute: no-padding
+	*/
+	@attr({ mode: 'boolean'}) checked = false;
+
+	/**
+	* Controls the selected state of the box
+	*
+	* @public
+	* HTML Attribute: no-padding
+	*/
+	@attr({ mode: 'boolean'}) clickable = false;
+
+	handleCheckedChange() {
```

Comment:
This (and keydown handler) needs internal annotation or it will be exported as part of the custom element manifest.
Additionally, it is preferable to prefix it with an underscore to discourage users from calling it anyway.
```suggestion
/**
 * @internal
 */
	_handleCheckedChange() {
```